### PR TITLE
Form support clearOnDestroy

### PR DESCRIPTION
**Pull request by crazyair**
[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |    Form support clearOnDestroy       |
| 🇨🇳 Chinese |   Form 支持 clearOnDestroy        |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

### CODE REVIEW

File: components/form/index.en-US.md [hunk 1]
```diff
@@ -88,6 +88,7 @@ Common props ref：[Common props](/docs/react/common-props)
 | onFinish | Trigger after submitting the form and verifying data successfully | function(values) | - |  |
 | onFinishFailed | Trigger after submitting the form and verifying data failed | function({ values, errorFields, outOfDate }) | - |  |
 | onValuesChange | Trigger when value updated | function(changedValues, allValues) | - |  |
+| clearOnDestroy | Clear form values when the form is uninstalled | boolean | false | 5.18.0 |
```
Comment (afc163) [inline components/form/index.en-US.md:91]: 这个和 preserve={false} 有啥区别？

Comment (crazyair) [reply components/form/index.en-US.md:91]: https://github.com/react-component/field-form/pull/692
Form 卸载清空 stroe 值，更适用于 Modal  + Form 形式，如果用 `preserve` 会丢失了当需要 `Form.Item` 卸载时保持值的特性

Comment (afc163) [reply components/form/index.en-US.md:91]: clearOnUnmount 是不是更好一点。

Comment (crazyair) [reply components/form/index.en-US.md:91]: clearOnUnmount 这个叫起来，好像要发出2个`昂`

**Comment by crazyair** [comment]
@SocketSecurity ignore npm/sharp@0.33.3

File: components/form/index.en-US.md (hunk 1 above)
Comment (afc163) [reply components/form/index.en-US.md:91]: 正是因为 destroyOnClose 和这个语义不一样，所以这里不太好用 destroy。destroy 是用于表达销毁 dom 元素，unmount 用于 react 卸载更合适。

Comment (crazyair) [reply components/form/index.en-US.md:91]: field-form 要改改吗

Comment (afc163) [reply components/form/index.en-US.md:91]: 改吧。

Comment (zombieJ) [reply components/form/index.en-US.md:91]: `unmount` 会引入新的 API 概念，这边 unmount 其实也是会销毁 dom 的。感觉可以不用新增概念。

**Review by zombieJ — APPROVED** [review]

**Pull request merged**